### DISCUSSION TITLE
TypeScript error in unit-test wrapper after upgrading to `v3.0.0`

### DISCUSSION THREAD

**Discussion by xsjcTony** [react-hook-form/react-hook-form#10160] [Q&A]
**Describe the bug**
Hey, the Type error happened after upgrading to `v3.0.0`, and I'm using generic to create a wrapper with `react-hook-form` in my unit test.
It's working as expected in `v2`

**To Reproduce**
Open the link below, the error will be there

**Stackblitz link**
https://stackblitz.com/edit/vitejs-vite-ylobmg?file=src/__tests__/Demo.test.tsx

**Expected behaviour**

**Screenshots**
![image](https://user-images.githubusercontent.com/45784210/226513441-27fab760-d492-4efb-a658-e468c56547e7.png)

**Desktop (please complete the following information):**

- OS: Windows
- Browser: Chrome
- Version: 111

**Comment by jorisre** [comment]
Heya 👋🏻 

Can you try this solution:

```tsx
import { yupResolver } from '@hookform/resolvers/yup';
import { useForm } from 'react-hook-form';
import * as yup from 'yup';
import type { ObjectSchema } from 'yup';
import type { InputHTMLAttributes } from 'react';

interface SchemaType1 {
  test: string;
}

interface WrapperWithHookFormProps<T extends ObjectSchema<any>>
  extends InputHTMLAttributes<HTMLInputElement> {
  schema: T;
}

const schema1: ObjectSchema<SchemaType1> = yup.object({
  test: yup.string().required(),
});

const WrapperWithHookForm = <T extends ObjectSchema<any>>({
  schema,
  ...props
}: WrapperWithHookFormProps<T>): JSX.Element => {
  const { register } = useForm<T>({ resolver: yupResolver(schema) });

  return <input {...props} />;
};
```

**Comment by xsjcTony** [comment]
@jorisre Hey, thanks for the solution, but I actually need that `K` generic since I need to constraint the `name` passed into `<input>` in later test cases (sorry it's not included in the minimal reproduction), so is there any solution that remain with generic T and K?

Since it's working in `v2`, no idea what breaks it in `v3`

**Comment by jorisre** [comment]
It probably breaks because of Yup v1 upgrade. Can you add more detail in your repro? To understand what you want to achieve?

**Comment by xsjcTony** [comment]
Sure. I'll update the reprod. I've been using `yup` v1 for a while, so that should not be the issue. Basically if you change the version of `@hookform/resolvers` in my reprod, the issue is gone, so that's why I'm pretty sure upgrade to `v3` breaks this.

**Comment by xsjcTony** [comment]
@jorisre I've updated the reprod (it's another forked one): https://stackblitz.com/edit/vitejs-vite-ylobmg?file=src/__tests__/Demo.test.tsx

**Comment by jorisre** [comment]
Moved your issue to discussion. It's not a react-hook-form|resolvers bug

**Reply by xsjcTony** [reply]
Sure. But is there any workaround for this? As I'm not really sure what's going wrong... like I said once I switch `@hookform/resolvers` to `v2` then the issue is gone. So if it has to be like that, I guess I'll need to ping the version to the latest `v2.x` since `v3` won't work.

**Reply by xsjcTony** [reply]
Just to update this, it's defeinitely a bug in `@hookform/resolvers`... since my issue is fixed by https://github.com/react-hook-form/resolvers/pull/563 here.
